### PR TITLE
feat: functionality

### PR DESCRIPTION
**Pull request by harshitadaswani**
- added like functionality
- added watch later functionality
- added history functionality

### CODE REVIEW

File: src/context/HistoryContext.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { createContext, useContext, useReducer } from "react";
+import { initialHistory, HistoryReducer } from "../reducer/HistoryReducer";
+
+const defaultHistoryValue = {};
+
+const HistoryContext = createContext(defaultHistoryValue);
```
Comment (paarthbajaj) [inline src/context/HistoryContext.jsx:7]: Can you please comment here, what was the use of "deafultHistoryValue" variable?

File: src/components/shared/card/HistoryCard.jsx [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
-            <h4 className="subtitle-basic-btns m-xs">{creator}</h4>
-            <Link to={`/player/${_id}`}>
-              <button className="bid-btn br-s p-s m-xs fw-semibold txt-s">
-                Watch Now
-              </button>
-            </Link>
-          </div>
-        );
-      })}
+      {historyState.history.length !== 0 ? (
```
Comment (paarthbajaj) [inline src/components/shared/card/HistoryCard.jsx]: Code quality is good and I liked that you have tried to cover the edge cases.

**Review by paarthbajaj — APPROVED** [review]
Code quality is great with the edge cases covered. You can merge this PR.

File: src/context/HistoryContext.jsx (hunk 1 above)
Comment (harshitadaswani) [reply src/context/HistoryContext.jsx:7]: It is not useful here but I was following the syntax so defined it. Will refactor it! thanks for pointing it out

**Review by Avin008 — APPROVED** [review]
clean && well-structured code, good to merge

**Pull request merged**